tools/lints/sotn-lint/src/player_status.rs

```diff
@@ -0,0 +1,168 @@
+        let result = DMT.transform_line(input_line);
+        assert_eq!(result, expected_line)
+    }
+
+    #[test]
+    fn test_player_status_zero() {
+        let input_line = "g_Player.status = 0;";
+        let expected_line = "g_Player.status = 0;";
+        let result = DMT.transform_line(input_line);
+        assert_eq!(result, expected_line)
+    }
+
+    #[test]
+    fn test_player_status_zero_hex() {
+        let input_line = "g_Player.status = 0x0;";
+        let expected_line = "g_Player.status = 0;";
+        let result = DMT.transform_line(input_line);
+        assert_eq!(result, expected_line)
+    }
+
+    #[test]
+    fn test_player_status_flags() {
+        let input_line = "g_Player.status = PLAYER_STATUS_AXEARMOR;";
+        let expected_line = "g_Player.status = PLAYER_STATUS_AXEARMOR;";
+        let result = DMT.transform_line(input_line);
+        assert_eq!(result, expected_line)
+    }
+
+    #[test]
+    fn test_player_status_set() {
+        let input_line = "g_Player.status |= 0x2000000;";
+        let expected_line = "g_Player.status |= PLAYER_STATUS_ABSORB_BLOOD;";
+        let result = DMT.transform_line(input_line);
+        assert_eq!(result, expected_line)
+    }
+
+    #[test]
+    fn test_player_status_clear() {
+        let input_line = "g_Player.status &= 0xFFFF7FFF;";
+        let expected_line = "g_Player.status &= ~PLAYER_STATUS_CURSE;";
+        let result = DMT.transform_line(input_line);
+        assert_eq!(result, expected_line)
+    }
+
+    #[test]
+    fn test_player_status_clear_many() {
+        let input_line = "g_Player.status &= 0xFFFF6FFF;";
+        let expected_line = "g_Player.status &= ~(PLAYER_STATUS_CURSE | PLAYER_STATUS_UNK1000);";
+        let result = DMT.transform_line(input_line);
+        assert_eq!(result, expected_line)
+    }
+
+    #[test]
+    fn test_player_status_inverted_many() {
+        let input_line = "g_Player.status &= ~0x300;";
+        let expected_line = "g_Player.status &= ~(PLAYER_STATUS_UNK200 | PLAYER_STATUS_UNK100);";
+        let result = DMT.transform_line(input_line);
+        assert_eq!(result, expected_line)
+    }
+
+    #[test]
+    fn test_equality() {
+        let input_line = "if (g_Player.status == 8) {";
+        let expected_line = "if (g_Player.status == PLAYER_STATUS_UNK8) {";
+        let result = DMT.transform_line(input_line);
+        assert_eq!(result, expected_line)
+    }
+
+    #[test]
+    fn test_inequality() {
+        let input_line = "if (g_Player.status != 8) {";
+        let expected_line = "if (g_Player.status != PLAYER_STATUS_UNK8) {";
+        let result = DMT.transform_line(input_line);
+        assert_eq!(result, expected_line)
+    }
+
+    // TODO: Implement PLAYER_STATUS_TRANSFORM 0x7 flag
+    //
+    // #[test]
+    // fn test_player_status_combo_flag() {
+    //     let input_line = "g_Player.status = 0x7;";
+    //     let expected_line = "g_Player.status = PLAYER_STATUS_TRANSFORM;";
```

Comment:
@hohle as mentioned on Discord, this enum value does not map cleanly in the bitflag. I've left it off this linter for now
https://github.com/Xeeynamo/sotn-decomp/blob/master/include/game.h#L283-L285

Not sure if you have any ideas on how to implement something like this into the BitFlagsTransformer nicely